compio-driver/src/key.rs

```diff
@@ -1,40 +1,160 @@
+use std::{
+    io,
+    marker::PhantomData,
+    ops::{Deref, DerefMut},
+    pin::Pin,
+    task::Waker,
+};
+
+use compio_buf::BufResult;
+
+use crate::{OpCode, Overlapped, PushEntry, RawFd};
+
+#[repr(C)]
+pub struct RawOp<T: ?Sized> {
+    header: Overlapped,
+    // The two flags here are manual reference counting. The driver holds the strong ref until it
+    // completes; the runtime holds the strong ref until the future is dropped.
+    cancelled: bool,
+    upcast_fn: unsafe fn(usize) -> *mut RawOp<dyn OpCode>,
+    result: PushEntry<Option<Waker>, io::Result<usize>>,
+    op: T,
+}
+
+impl<T: ?Sized> RawOp<T> {
+    pub fn as_op_pin(&mut self) -> Pin<&mut T> {
```

Comment:
This is NOT safe. Pin contract requires the pointed value is pinned even after `'1` is ended here, until the data is dropped.

Reply:
It would be better if we can add a function on `Key<T>` to get a pinned op, because in context of `Key`, we know that the op is pinned on box and can be safely used.